### ISSUE TITLE
libbfd-2.26-system.so

### ISSUE THREAD

**Issue by peekaa** [vitalif/grive2#85]
After last update via deb package for Debian Sid, I got this error:
`grive: error while loading shared libraries: libbfd-2.26-system.so: cannot open shared object file: No such file or directory`

`

> root@mybox# apt show grive
> Package: grive
> Version: 0.5-1+git20160711~webupd8~xenial
> Priority: optional
> Section: net
> Source: grive2
> Maintainer: Alin Andrei <webupd8@gmail.com>
> Installed-Size: 1 792 kB
> Depends: binutils (>= 2.26), binutils (<< 2.27), libboost-filesystem1.58.0, libboost-program-options1.58.0, libboost-regex1.58.0, libboost-system1.58.0, libc6 (>= 2.14)
> Download-Size: 315 kB
> APT-Manual-Installed: yes
> APT-Sources: http://ppa.launchpad.net/nilarimogard/webupd8/ubuntu xenial/main amd64 Packages
> Description: Grive2: an open source Linux client for Google Drive
> options1.58.0, libboost-regex1.58.0, libboost-system1.58.0, libc6 (>= 2.14)
> Download-Size: 315 kB
> APT-Manual-Installed: yes
> APT-Sources: http://ppa.launchpad.net/nilarimogard/webupd8/ubuntu xenial/main amd64 Packages
> Description: Grive2: an open source Linux client for Google Drive

`

**Comment by cyphos1973** [comment]
I have the same error. I wait the fix bug.
Thanks you

**Comment by vitalif** [comment]
This error is build-related, and since I'm not the author of that PPA I can't fix it there.
The package built with `dpkg-buildpackage` ran from git repository has correct dependencies, so you can try to build it yourself (just download the git version and run `dpkg-buildpackage`)

**Comment by peekaa** [comment]
OK, I have just wrote to the Maintainer Alin. Hope it will be fixed soon. I thing many users use this PPA.

**Comment by hotice** [comment]
Hi, I'm Andrew, the maintainer of the grive2 package in the WebUpd8 PPA. This doesn't appear to be a dependency issue. A similar issue was present in the old Grive too: https://bugs.launchpad.net/ubuntu/+source/grive/+bug/1155194 and it looks like it was occurring because the package was old, built against old libraries.

@peekaa Since you're using Debian, I'm guessing some dependency (probably binutils) was updated in Debian and since Launchpad PPAs only support Ubuntu (well, some work in Debian, but then issues like this occur), I can't fix it for Debian. You may need to compile this yourself.

**Comment by vitalif** [comment]
Yes, it's about binutils updated from 2.26 to 2.26.1, but at least when I build it in debian it gets a specific version of binutils in dependencies (>=2.26.1 <2.26.2) which would prevent this bug from appearing. However I'm not sure if it was so with binutils 2.26...

**Comment by cyphos1973** [comment]
This is the solution
sudo ln -s /usr/lib/i386-linux-gnu/libbfd-2.26.1-system.so /usr/lib/i386-linux-gnu/libbfd-2.26-system.so

Now, it works

Thanks you a lot

**Comment by peekaa** [comment]
@hotice cyphos1973 is using Ubuntu https://github.com/vitalif/grive2/issues/85#issuecomment-232419941, so the issue might occurs even there after last update, so it might be worth fixing... Thank you.

**Comment by hotice** [comment]
Indeed, I just got the binutils update in Ubuntu too, so I guess a simple rebuild should fix it. Expect an update in about an hour or so (depends on how fast Launchpad can build the packages).

**Comment by cyphos1973** [comment]
after a few minutes, this is the error I don't know why
`cyphos@LENOVO-X230:~/Google_drive$ sudo grive
[sudo] Mot de passe de cyphos :          
Reading local directories
exception: /build/grive2-0T_JJ9/grive2-0.5/libgrive/src/util/OS.cc(57): Throw in function void gr::os::Stat(const string&, gr::DateTime*, off_t*, bool*)
Dynamic exception type: boost::exception_detail::clone_impl<gr::os::Error>
[gr::expt::BacktraceTag*] = #0 0x8187d7a grive gr::Exception::Exception()
#1 0x818271e grive gr::os::Error::Error()
#2 0x8182334 grive gr::os::Stat(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, gr::DateTime*, long long*, bool*)
#3 0x818222c grive gr::os::Stat(boost::filesystem::path const&, gr::DateTime*, long long*, bool*)
#4 0x814cf74 grive gr::Resource::FromLocal(gr::Val&)
#5 0x8136cfa grive gr::State::FromLocal(boost::filesystem::path const&, gr::Resource*, gr::Val&)
#6 0x8136da0 grive gr::State::FromLocal(boost::filesystem::path const&, gr::Resource*, gr::Val&)
#7 0x8136da0 grive gr::State::FromLocal(boost::filesystem::path const&, gr::Resource*, gr::Val&)
#8 0x8136da0 grive gr::State::FromLocal(boost::filesystem::path const&, gr::Resource*, gr::Val&)
#9 0x81367cb grive gr::State::FromLocal(boost::filesystem::path const&)
#10 0x81555f6 grive gr::Drive::DetectChanges()
#11 0x811abb9 grive Main(int, char**)
#12 0x811b38c grive main
#13 0xb6f71637 /lib/i386-linux-gnu/libc.so.6 __libc_start_main
#14 0x81193d1 grive 

2, "No such file or directory"[boost::errinfo_file_name_*] = ./PROJETS/Symfony-clean-architecture/bin/doctrine
[boost::errinfo_api_function_*] = stat`

**Comment by peekaa** [comment]
in my debian it is:  ln -s /usr/lib/x86_64-linux-gnu/libbfd-2.26.1-system.so /usr/lib/x86_64-linux-gnu/libbfd-2.26-system.so

**Comment by peekaa** [comment]
@cyphos1973 dont run grive as root/sudo...

**Comment by cyphos1973** [comment]
@peekaa 
I tried the both
sudo grive or grive, it's always the same error

**Comment by hotice** [comment]
@cyphos1973 the PPA should be updated in a few minutes (I already updated it but Launchpad takes a while to publish the packages) with rebuilt packages, which should fix this.

**Comment by vitalif** [comment]
> 2, "No such file or directory"[boost::errinfo_file_name_] = ./PROJETS/Symfony-clean-architecture/bin/doctrine
[boost::errinfo_api_function_] = stat`

No such file or directory: ./PROJETS/Symfony-clean-architecture/bin/doctrine

Did you delete the file during sync? :)

**Comment by vitalif** [comment]
> in my debian it is: ln -s /usr/lib/x86_64-linux-gnu/libbfd-2.26.1-system.so /usr/lib/x86_64-linux-gnu/libbfd-2.26-system.so

this is NOT a good solution, you should better rebuild it from git...

**Comment by peekaa** [comment]
@vitalif ok, reverted
@hotice works!

**Comment by cyphos1973** [comment]
@vitalif 
no, i didn't delete the file during the sync...
I tried some times without modify the files during the sync. So i watched this directory
`lrwxrwxrwx 1 cyphos www-data 35 nov.   4  2015 PROJETS/Symfony-blog-v1/bin/doctrine -> ../vendor/doctrine/orm/bin/doctrine
`
I think the link symbolic would be a problem, is not?
